File: 800 rated/HalloumiBoxes/HalloumiBoxes.cpp
#include <bits/stdc++.h>
using namespace std;
int isSorted(vector<long long> a){
  int n = a.size();
  for(int i = 1; i<n;i++){
    if(a[i]<a[i-1]) return -1;
  }
  return 1;
}
int main(){
  int t;
  cin>>t;
  while(t--){
    long long n,k;
    cin>>n>>k;
    vector<long long>a(n);
    for(int i = 0;i<n;i++){
      cin>>a[i];
    }
    if(isSorted(a)==1||k>1) cout<<"YES"<<endl;
    else cout<<"NO"<<endl;
  }
  return 0;
}

File: 800 rated/HalloumiBoxes/HalloumiBoxes.java
import java.io.*;
import java.util.*;

class HalloumiBoxes {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        PrintWriter out = new PrintWriter(new BufferedWriter(new OutputStreamWriter(System.out)));

        int T = Integer.parseInt(br.readLine());
        while (T-- > 0) {

            StringTokenizer st = new StringTokenizer(br.readLine());
            int n = Integer.parseInt(st.nextToken());
            int k = Integer.parseInt(st.nextToken());

            int[] arr = new int[n];
            st = new StringTokenizer(br.readLine());

            for (int i = 0; i < n; i++) {
                arr[i] = Integer.parseInt(st.nextToken());
            }

            boolean isSorted = true;
            for (int i = 0; i < n - 1; i++) {
                if (arr[i] > arr[i + 1]) {
                    isSorted = false;
                    break;
                }
            }

            if (k == 1 && !isSorted) out.println("NO");
            else out.println("YES");
        }

        out.flush();
    }
}